routes/user.py

```diff
@@ -1,22 +1,69 @@
-from fastapi import APIRouter
-from models.user import User
-from schemas.user import users_serializer
+from bson import ObjectId
+from fastapi import APIRouter, HTTPException, status, Request
+from db.models import users_serializer
+from schemas.user import User
 from config.db import collection
+from db.hash import Hash
+from jose import jwt
+from utilities.helper import remove_field_document
+from fastapi import APIRouter, Depends
+from starlette.middleware.base import BaseHTTPMiddleware
+
+from middelware.auth import auth_middleware
 
 
 user = APIRouter(prefix="/user", tags=['user'])
 
-@user.post("")
+
+@user.post("/register")
 async def create_user(user: User):
+    existing_user = collection.find_one({"email": user.email})
+    if existing_user:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="User already exists")
+    
+    hashed_password = Hash.bcrypt(user.password)
+    user.password = hashed_password
+    
     _id = collection.insert_one(dict(user))
     user = users_serializer(collection.find({"_id": _id.inserted_id}))
     return {"status": "Ok","data": user}
 
+
+@user.post("/login")
+async def login_user(user: User):
+    found_user = collection.find_one({"email": user.email})
+    
+    if not found_user:
+        raise HTTPException(status_code=400, detail="User not found")
```

Comment:
I would stick to one style only, i.e. status.HTTP_400_BAD_REQUEST rather than 400, though it output is the same.

Reply:
I have done it